### PR TITLE
Removed Protobuf2 serialization from Akka.Persistence

### PR DESCRIPTION
**Pull request by alexvaluyskiy**
- Removed `MessageSerializer`
- Removed `SnapshotSerializer`
- Removed Google.ProtocolBuffer dependency and PCL compatibility mode

It could introduce some breaking changes in our plugins because previously all `AtLeastOnceDeliverySnapshots` used `protobuf2` serializer

### CODE REVIEW

File: src/contrib/cluster/Akka.Cluster.Sharding.Tests/Akka.Cluster.Sharding.Tests.csproj [hunk 1]
```diff
@@ -4,7 +4,6 @@
   <PropertyGroup>
     <AssemblyTitle>Akka.Cluster.Sharding.Tests</AssemblyTitle>
     <TargetFrameworks>net452;netcoreapp1.1</TargetFrameworks>
-    <PackageTargetFallback Condition=" '$(TargetFramework)' == 'netcoreapp1.1' ">$(PackageTargetFallback);portable-net45+win8</PackageTargetFallback>
```
Comment (alexvaluyskiy) [inline src/contrib/cluster/Akka.Cluster.Sharding.Tests/Akka.Cluster.Sharding.Tests.csproj]: Removed all `PackageTargetFallback` from Persistence projects.

File: src/core/Akka.Persistence.Tests/Serialization/SerializerSpec.cs [hunk 2]
```diff
@@ -204,7 +204,7 @@ public override string ToString()
         }
     }*/
 
-    public class SnapshotSerializerPersistenceSpec : AkkaSpec
+    public abstract class SnapshotSerializerPersistenceSpec : AkkaSpec
```
Comment (alexvaluyskiy) [inline src/core/Akka.Persistence.Tests/Serialization/SerializerSpec.cs]: Test is skipped temporarily

File: src/core/Akka.Persistence.Tests/Serialization/SerializerSpec.cs [hunk 3]
```diff
@@ -242,7 +242,7 @@ public void SnapshotSerializer_should_handle_custom_snapshot_Serialization_with_
         }
     }
 
-    public class MessageSerializerPersistenceSpec : AkkaSpec
+    public abstract class MessageSerializerPersistenceSpec : AkkaSpec
```
Comment (alexvaluyskiy) [inline src/core/Akka.Persistence.Tests/Serialization/SerializerSpec.cs]: Test is skipped temporarily

File: src/core/Akka.Persistence.Tests/Serialization/SerializerSpec.cs [hunk 4]
```diff
@@ -358,7 +358,7 @@ public void MessageSerializer_when_given_AtLeastOnceDeliverySnapshot_should_hand
         }
     }
 
-    public class MessageSerializerRemotingSpec : AkkaSpec
+    public abstract class MessageSerializerRemotingSpec : AkkaSpec
```
Comment (alexvaluyskiy) [inline src/core/Akka.Persistence.Tests/Serialization/SerializerSpec.cs]: Test is skipped temporarily

File: src/core/Akka.Persistence.Tests/Serialization/SerializerSpec.cs (hunk 2 above)
Comment (alexvaluyskiy) [inline src/core/Akka.Persistence.Tests/Serialization/SerializerSpec.cs]: Will move these specs to `TCK`

**Comment by Danthar** [comment]
Is this still WIP? Because it looks done.

**Comment by alexvaluyskiy** [comment]
@Danthar @heynickc We can't merge any persistence pull requests because we muted all Akka.Persistence.Tests tests

**Comment by Aaronontheweb** [comment]
@alexvaluyskiy need to resolve conflicts here before we can move forward with merging

**Comment by Aaronontheweb** [comment]
@alexvaluyskiy nice work

**Pull request merged**